micro-rdk/src/common/board.rs

```diff
@@ -85,6 +85,22 @@ pub trait Board: DoCommand {
         ))
     }
 
+    #[cfg(feature = "esp32")]
+    fn add_digital_interrupt_callback(
+        &mut self,
+        _pin: i32,
+        _intr_type: crate::esp32::esp_idf_svc::hal::gpio::InterruptType,
+        _cb: crate::esp32::esp_idf_svc::hal::sys::gpio_isr_t,
+        _arg: Option<*mut core::ffi::c_void>,
```

Comment:
I considered making this a struct, but it complicated things without adding much value.
While we'd be able to provide a sane default for the callback, we wouldn't be able to do the same for the callback arg